@here/harp-mapview/lib/Utils.ts

```diff
@@ -481,3 +482,107 @@ export namespace MapViewUtils {
         return language.substring(0, 2);
     }
 }
+
+export namespace TileOffsetUtils {
+    /**
+     * Creates a unique key based on the supplied parameters. Note, the uniqueness is bounded by the
+     * bitshift. The mortonCode supports currently up to 26 levels, the bitshift reduces this
+     * accordingly, so given the default bitshift of four, we support up to 22 levels. Given the
+     * current support up to level 19 this should be fine.
+     *
+     * @param tileKey The unique [[TileKey]] from which to compute the unique key.
+     * @param offset How much the given [[TileKey]] is offset
+     * @param bitshift How much space we have to store the offset. The default of 4 means we have
+     *      enough space to store 16 unique tiles.
+     */
+    export function getKeyForTileKeyAndOffset(
+        tileKey: geoUtils.TileKey,
+        offset: number,
+        bitshift: number = 3
+    ) {
+        const shiftedOffset = getShiftedOffset(offset, bitshift);
+        return tileKey.mortonCode() + shiftedOffset;
+    }
+
+    /**
+     * Packs the supplied offset into the high bits, where the highbits are between 2^53 and
+     * 2^(53-bitshift+1), +1 because we use one bit for storing if the given offset is negative.
+     *
+     * Whether the offset is negative is stored in the bit before the `bitshift` high bits.
+     *
+     * The given offset has the modulo taken of 2^bitshift incase it is bigger than this number.
+     *
+     * Note, because bit shifting in JavaScript works on 32 bit integers, we use powers of 2 to set
+     * the high bits instead.
+     *
+     * @param offset Offset to pack into the high bits.
+     * @param bitshift How many bits to use to pack the offset.
+     */
+    export function getShiftedOffset(offset: number, bitshift: number = 3) {
+        let result = 0;
+        const maxOffset = Math.pow(2, bitshift);
```

Comment:
Why not using bitwise shidting operator `<<` which is 2x times faster rather than `Math.pow(2, x)` ? 
http://jsfiddle.net/awncj81m/

Reply:
`<<` changes every number (double precision float) to a 32 bit unsigned integer, which is maybe not what we want here. In geoutils' `TileKey` I have some workaround for the slow Math.pow while still keeping the 53 bits precision of double

Reply:
@haraldF, thanks, yes that would be good, I will take a look.
@sasha240100 , I will add a comment to make it clear for those that aren't of Javascripts bitshift eccentricities

Reply:
sorry @sasha240100 , this shouldn't be reviewed, hence I have closed it, however the code in particular you reviewed, this I will keep, so hopefully soon I will have something to review.